### PR TITLE
Build tooling update 

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
- `make apply` is not longer needed thanks to go embed
- `make server` now only build the current `OS` & `ARCH` combination if `MM_SERVICESETTINGS_ENABLEDEVELOPER` is set allowing developers to build faster
- Removed some unuseful consts and add new ones
- Dependency update

#### Ticket Link
None

### CODE REVIEW

**Review by sibasankarnayak — APPROVED** [review]
LGTM!

File: .golangci.yml [hunk 1]
```diff
@@ -15,6 +12,8 @@ linters-settings:
   govet:
     check-shadowing: true
     enable-all: true
+    disable:
+      - fieldalignment
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .golangci.yml:16]: It's a low value linter (of us) which causes harder to read code. See https://github.com/mattermost/mattermost-plugin-playbooks/pull/505 for more details.

File: build/custom.mk [hunk 2]
```diff
@@ -1 +1,2 @@
 # Include custom targets and environment variables here
+default: all
```
[the remark this answers is not in the document]
Comment (hanzei) [reply build/custom.mk:2]: It's already defined there: https://github.com/mattermost/mattermost-plugin-github/blob/f9e1c9418ed3b045f2212ad7bd140b0832d65d5a/Makefile#L19-L21

Removing the line.

File: Makefile [hunk 3]
```diff
@@ -57,30 +55,37 @@ ifneq ($(HAS_SERVER),)
...
-## Builds the server, if it exists, for all supported architectures.
+## Builds the server, if it exists, for all supported architectures, unless MM_SERVICESETTINGS_ENABLEDEVELOPER is set
 .PHONY: server
 server:
 ifneq ($(HAS_SERVER),)
 	mkdir -p server/dist;
 ifeq ($(MM_DEBUG),)
+ifneq ($(MM_SERVICESETTINGS_ENABLEDEVELOPER),)
+	cd server && $(GO) build $(GO_BUILD_FLAGS) -trimpath -o dist/plugin-$(DEFAULT_GOOS)-$(DEFAULT_GOARCH);
+else
```
[the remark this answers is not in the document]
Comment (hanzei) [reply Makefile:66]: Sounds reasonable to me. Let's do that in https://github.com/mattermost/mattermost-plugin-starter-template to unify.

[the remark this answers is not in the document]
Comment (hanzei) [reply Makefile:66]: https://github.com/mattermost/mattermost-plugin-starter-template/pull/161

**Pull request merged**